hive-bigquery-connector-common/src/main/java/com/google/cloud/hive/bigquery/connector/sparksql/SparkSQLOutputCommitter.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.hive.bigquery.connector.sparksql;
+
+import com.google.cloud.hive.bigquery.connector.output.BigQueryOutputCommitter;
+import com.google.cloud.hive.bigquery.connector.output.OutputCommitterUtils;
+import java.io.IOException;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.JobStatus;
+import org.apache.hadoop.mapreduce.OutputCommitter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+/** Output committer compatible with the new "mapreduce" Hadoop API. */
+public class SparkSQLOutputCommitter extends OutputCommitter {
+
+  @Override
+  public void commitJob(org.apache.hadoop.mapreduce.JobContext jobContext) throws IOException {
+    SparkSQLUtils.cleanUpSparkJobFile(jobContext.getConfiguration());
```

Comment:
can this spark job file be moved to inside per query? right now it is per spark application, in case a previous query failure or the committer not called, this existing file will cause problem for next query?

Reply:
So this Spark job file only contains one piece of information: what tables are "INSERT" and what tables are "INSERT OVERWRITES". So I think it's relevant for the whole Spark application. Makes sense?

Reply:
The Spark job file basically mimics what Hive itself does here: https://github.com/apache/hive/blob/8190d2be7b7165effa62bd21b7d60ef81fb0e4af/ql/src/java/org/apache/hadoop/hive/ql/parse/QBParseInfo.java#L69-L70

Reply:
this works if the spark application fails when the insert query fails. but if user choose to handle the spark sql failure and continue the spark application (maybe a try catch), or user run in spark-shell, the same spark application can continue, so later another insert query (either insert into same table or different table), what would happen?

Reply:
In that case I think the spark job file would be overwritten when a new query is run.

BTW, if you run multiple queries as part of the same Spark shell session, would the spark app ID be the same for all queries, would they be unique?

Reply:
they will have the same application id. 
the logic looks like readOrElseCreate the spark job json file for each query?

Reply:
You're right, I was using some readOrElseCreate logic because somehow Spark calls the `apply()` method multiple times. I've just changed it to always write the file. That way it should always be overwritten. What do you think?

Reply:
would that cause locking issue if there are concurrent writes to this file? Is there difficulty of moving this under the temporary query directory (or is the temporary query directory is not available at this time)?

Reply:
@yigress  What do you mean by "temporary query directory"? Are you referring to the directory named after `hive.query.id` when using plain Hive? If so, as far as I can tell, there's no way to retrieve a unique ID for the query provided in the query plan object in the Spark extension strategy's `apply()` function. Or is there?

The challenge here is that we'd need some unique way of identifying the query that would be deterministically created or retrieved both from the Spark extension and from the Hive storage handler.

For now, the only thing I'm aware of is the spark app ID. But as you mentioned, that's not enough to identify a specific query.

Another question for you: If you run multiple queries as part of the same Spark Shell session (i.e. the same Spark App ID), could multiple queries be run in parallel, or could they only be run in sequence? If in sequence, then perhaps the Spark App ID is enough. But if in parallel, then we're in trouble :)